aom/src/main/java/com/nedap/archie/aom/utils/ArchetypeParsePostProcesser.java

```diff
@@ -51,12 +51,27 @@ private static void setParents(Archetype archetype) {
                 workList.addAll(attribute.getChildren());
             }
             if(cObject instanceof CComplexObject) {
+                //TODO: fix tuple so it has proper multiple references to the separate structures, instead of a complete duplication of data in json
                 CComplexObject cComplexObject = (CComplexObject) cObject;
                 for(CAttributeTuple tuple: cComplexObject.getAttributeTuples()) {
                     for(CAttribute attribute:tuple.getMembers()) {
                         attribute.setSocParent(tuple);
+                        attribute.setParent(cObject);
+                        workList.addAll(attribute.getChildren());
                         cComplexObject.replaceAttribute(attribute);
                     }
+                    for(CPrimitiveTuple primitiveTuple:tuple.getTuples()) {
+                        int index = 0;
+                        for(CPrimitiveObject object:primitiveTuple.getMembers()) {
+                            if(index < tuple.getMembers().size()) {
+                                CAttribute attribute = tuple.getMember(index);
+                                object.setSocParent(tuple);
```

Comment:
I'm not sure what exactly the SocParent does, but in https://github.com/openEHR/archie/blob/ef60904f0b195afa8b242e76c70f765de7239127/tools/src/main/java/com/nedap/archie/flattener/TupleFlattener.java#L39 it is set to the CPrimitiveTuple instead of the CAttributeTuple

Reply:
Very poorly specified in https://specifications.openehr.org/releases/AM/latest/AOM2.html#_second_order_constraints . But I'll change it to the same as there, thanks